libs/ipc/inc/public/tfc/ipc/details/dbus_server_iface.hpp

```diff
@@ -248,18 +259,39 @@ class ipc_manager_client {
         std::bind_front(&ipc_manager_client::match_callback, this));
   }
 
+  /**
+   * Register a signal with the ipc_manager service running on dbus
+   * @tparam message_handler the type of the message handler should only accept a std::error_code reference
```

Comment:
I would recommend using `std::invocable<std::error_code> auto&& handler` instead in the function signature